server/routes/journal.py

```diff
@@ -101,12 +101,29 @@ def add_journals():
     return jsonify({"message": "Journal added successfully!", "status": 200}, body)
 
 
-@app.route("/api/journals/<identifier>", methods=["GET"])
+@app.route("/api/journal/<issn>", methods=["GET"])
 @cross_origin()
-def list_journal(identifier):
+def get_journal_by_issn(issn):
     """Lists general information from a journal, including its domains"""
-    return STUB_PAGE_MESSAGE
-
+    journal = Journal.query.filter_by(issn =issn).one()
+    res = []
+    issn = journal.issn
+    policy = Policies.query.filter_by(issn=issn)
+    domain = Domain.query.filter_by(issn=issn).first()
+    data = {
+            "issn": issn,
+            "title": journal.title,
+            "url": journal.url,
+            "rating": journal.ratings,
+            "policy_title": policy[0].title,
+            "first_year": policy[0].first_year,
+            "last_year": policy[0].last_year,
+            "policy_type": policy[0].policy_type,
+            "domain": domain.name,
+        }
+    res.append(data)
+    return jsonify({"message": "Journal found","status": 200, "data": res})
```

Comment:
About that `res` variable, some tips:

1. Instead of initializing `res` to an empty list just to later `append` a single element, you could simply do:

```python
    res = [{
        "issn": issn,
        ...
    }]
```

2. Since the field `"data"` only contains a single element, you could instead use `data` directly:

```suggestion
    issn = journal.issn
    policy = Policies.query.filter_by(issn=issn)
    domain = Domain.query.filter_by(issn=issn).first()
    data = {
            "issn": issn,
            "title": journal.title,
            "url": journal.url,
            "rating": journal.ratings,
            "policy_title": policy[0].title,
            "first_year": policy[0].first_year,
            "last_year": policy[0].last_year,
            "policy_type": policy[0].policy_type,
            "domain": domain.name,
        }
    return jsonify({"message": "Journal found","status": 200, "data": data})
```

- `res` won't be necessary
- According to my local tests, I've got the same results both with `res` as 1 element list and with `data` directly, which is:

```json
{
  "data": {
    "domain": "Biology",
    "first_year": 2015,
    "issn": 1234,
    "last_year": 2021,
    "policy_title": "Policy 1",
    "policy_type": "Requires",
    "rating": "A",
    "title": "Journal 1234",
    "url": "journal1234.com"
  },
  "message": "Journal found",
  "status": 200
}
```